### PR TITLE
[#3184] Add Axios definition for Flow v0.93

### PR DESCRIPTION
**Pull request by julianguyen**
- Type of contribution: new definition

Other notes:

Issue: https://github.com/flow-typed/flow-typed/issues/3184

Add Axios definition for Flow v0.94 to resolve issue with `http$ClientRequest`.

### CODE REVIEW

File: definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
+declare module 'axios' {
```
Comment (pascalduez) [inline definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js]: Please lower the scope of one version,  the breaking change was introduced in Flow 0.93.  
`axios_v0.18.x/flow_v0.93.x-/axios_v0.18.x.js`

File: definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    Axios: typeof Axios;
+    Cancel: Class<Cancel>;
+    CancelToken: Class<CancelToken>;
+    isCancel(value: any): boolean;
+    create(config?: AxiosXHRConfigBase<any, any>): Axios;
+    all: typeof Promise.all;
+    spread(callback: Function): (arr: Array<any>) => Function;
+  }
+  declare module.exports: AxiosExport;
+}
```
Comment (pascalduez) [inline definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js]: You'll need to also copy the tests, and scope them under the the `axios_v0.18.x/flow_v0.93.x-` folder.

File: definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js (hunk 1 above)
Comment (pascalduez) [reply definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js]: Ah humm sorry, and also re-scope the previous one. 

Rename without change
 `axios_v0.18.x/flow_v0.80.x-/axios_v0.18.x.js` --->  `axios_v0.18.x/flow_v0.80.x-v0.92.x/axios_v0.18.x.js`

New file with fixed definitions
 `axios_v0.18.x/flow_v0.93.x-/axios_v0.18.x.js`

File: definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js (hunk 2 above)
Comment (julianguyen) [reply definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js]: I notice `definitions/npm/axios_v0.18.x/test_axios_v0.18.x.js` exists so is it necessary to copy the tests?

Comment (pascalduez) [reply definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js]: Clarification: only if they break and requires some changes as well.

Comment (pascalduez) [reply definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js]: So let's wait for the CI and see :)

Comment (julianguyen) [reply definitions/npm/axios_v0.18.x/flow_v0.94.x-/axios_v0.18.x.js]: Looks like we don't need to copy tests then? :D

**Pull request merged**